failed to make

./configure.py
writing to '/home/kzhou/src/metag/pbdagcon/defines.mk'
"""
DALIGNER_INCLUDE     ?= ${DALIGNER_SRC}
DALIGNER_SRC         ?= /home/kzhou/src/metag/pbdagcon/../DALIGNER
DAZZ_DB_INCLUDE      ?= ${DAZZ_DB_SRC}
DAZZ_DB_SRC          ?= /home/kzhou/src/metag/pbdagcon/../DAZZ_DB
HDF5_LIBFLAGS        ?= -lhdf5_cpp -lhdf5
HTSLIB_LIBFLAGS      ?= -lhts
LIBBLASR_LIBFLAGS    ?= -lblasr
LIBPBDATA_LIBFLAGS   ?= -lpbdata
LIBPBIHDF_LIBFLAGS   ?= -lpbihdf
PBBAM_LIBFLAGS       ?= -lpbbam
ZLIB_LIBFLAGS        ?= -lz
"""
kzhou@jinshou:~/src/metag/pbdagcon$ make
make -C src/cpp
make[1]: Entering directory `/home/kzhou/src/metag/pbdagcon/src/cpp'
g++ -O3 -std=c++11 -Wall -Wuninitialized -pedantic -MMD -MP -I/home/kzhou/src/metag/pbdagcon/../DAZZ_DB -I/home/kzhou/src/metag/pbdagcon/../DALIGNER -isystem.//third-party  -c -o Alignment.o Alignment.cpp
g++ -O3 -std=c++11 -Wall -Wuninitialized -pedantic -MMD -MP -I/home/kzhou/src/metag/pbdagcon/../DAZZ_DB -I/home/kzhou/src/metag/pbdagcon/../DALIGNER -isystem.//third-party  -c -o AlnGraphBoost.o AlnGraphBoost.cpp
g++ -O3 -std=c++11 -Wall -Wuninitialized -pedantic -MMD -MP -I/home/kzhou/src/metag/pbdagcon/../DAZZ_DB -I/home/kzhou/src/metag/pbdagcon/../DALIGNER -isystem.//third-party  -c -o BlasrM5AlnProvider.o BlasrM5AlnProvider.cpp
g++ -O3 -std=c++11 -Wall -Wuninitialized -pedantic -MMD -MP -I/home/kzhou/src/metag/pbdagcon/../DAZZ_DB -I/home/kzhou/src/metag/pbdagcon/../DALIGNER -isystem.//third-party  -c -o main.o main.cpp
main.cpp:59:35: fatal error: tuples/TupleMetrics.hpp: No such file or directory
 #include "tuples/TupleMetrics.hpp"
                                   ^
compilation terminated.
make[1]: *** [main.o] Error 1
make[1]: Leaving directory `/home/kzhou/src/metag/pbdagcon/src/cpp'
make: *** [cpp] Error 2

Looks the make file is missing some library reference:
g++ -Wl,--no-as-needed -o pbdagcon Alignment.o AlnGraphBoost.o BlasrM5AlnProvider.o main.o SimpleAligner.o  -L/home/kzhou/lib -L/usr/local/lib -lblasr -lpbdata
/home/kzhou/lib/libblasr.so: undefined reference to `vtable for HDFData'
/home/kzhou/lib/libblasr.so: undefined reference to `H5::H5File::~H5File()'
/home/kzhou/lib/libblasr.so: undefined reference to `H5::StrType::StrType(int, unsigned long const
...  A lots more not shown

After adding -lhdf5_cpp -lhdf5, I am still getting more linking errors:
g++ -Wl,--no-as-needed -o pbdagcon Alignment.o AlnGraphBoost.o BlasrM5AlnProvider.o main.o SimpleAligner.o  -L/home/kzhou/lib -L/usr/local/lib -lhdf5_cpp -lhdf5 -lblasr -lpbdata
/home/kzhou/lib/libblasr.so: undefined reference to `vtable for HDFData'
/home/kzhou/lib/libblasr.so: undefined reference to `HDFNewBasReader::Close()'
/home/kzhou/lib/libblasr.so: undefined reference to `HDFAtom<std::string>::Read(std::string&)'
/home/kzhou/lib/libblasr.so: undefined reference to `vtable for HDFGroup'
/home/kzhou/lib/libblasr.so: undefined reference to `HDFData::InitializeDataset(HDFGroup&, std::st

After running travis.sh got different error messages:

/boost_1_58_0-headersonly -isystem.//third-party  -c -o dazcon.o dazcon.cpp
/home/kzhou/src/metag/pbdagcon/DALIGNER/align.c:61:16: fatal error: DB.h: No such file or directory
 #include "DB.h"
                ^
compilation terminated.
/home/kzhou/src/metag/pbdagcon/DAZZ_DB/DB.c: In function ‘Load_QVs’:
/home/kzhou/src/metag/pbdagcon/DAZZ_DB/DB.c:814:15: warning: ignoring return value of ‘fscanf’, declared with attribute warn_unused_result [-Wunused-result]
         fscanf(istub,DB_NFILE,&nfiles);
               ^
In file included from DazAlnProvider.cpp:45:0:
DazAlnProvider.hpp:52:16: fatal error: DB.h: No such file or directory
 #include "DB.h"
                ^
/home/kzhou/src/metag/pbdagcon/DAZZ_DB/DB.c:818:19: warning: ignoring return value of ‘fscanf’, declared with attribute warn_unused_result [-Wunused-result]
           { fscanf(istub,DB_FDATA,&last,fname,prolog);
                   ^
compilation terminated.
/home/kzhou/src/metag/pbdagcon/DAZZ_DB/DB.c:823:19: warning: ignoring return value of ‘fscanf’, declared with attribute warn_unused_result [-Wunused-result]
           { fscanf(istub,DB_FDATA,&last,fname,prolog);
                   ^
 File `all' does not exist.
   File `dazcon' does not exist.
make[1]: *** [align.o] Error 1
make[1]: *** Waiting for unfinished jobs....
make[1]: *** [DazAlnProvider.o] Error 1
In file included from dazcon.cpp:57:0:
DazAlnProvider.hpp:52:16: fatal error: DB.h: No such file or directory
 #include "DB.h"
                ^
compilation terminated.
make[1]: *** [dazcon.o] Error 1
make[1]: Leaving directory `/home/kzhou/src/metag/pbdagcon/src/cpp'
make: *** [cpp] Error 2

I do see an error, but not yours, and only after building.
```
$ make check
make -C test/cpp
make[1]: Entering directory 'pbdagcon/test/cpp'
make check
make[2]: Entering directory `pbdagcon/test/cpp'
g++ -O3 -std=c++11 -I. -I./ -Ipbdagcon/blasr_libcpp/alignment -Ipbdagcon/blasr_libcpp/pbdata -Ipbdagcon/blasr_libcpp/hdf -Ipbdagcon/src/cpp/third-party/boost_1_58_0-headersonly -Ipbdagcon/test/cpp/gtest-1.7.0/include -Ithird-party  -c -o TargetHitTest.o TargetHitTest.cpp
TargetHitTest.cpp:2:30: fatal error: DazAlnProvider.hpp: No such file or directory
 #include <DazAlnProvider.hpp>
```
I can fix that, but I'll need more info to solve *your* problem.

After some manual work, still not fully there yet. Any clue?

g++ -Wl,--no-as-needed -o pbdagcon Alignment.o AlnGraphBoost.o BlasrM5AlnProvider.o main.o SimpleAligner.o  -L/home/kzhou/lib -L/usr/local/lib -lhdf5_cpp -lhdf5 -lpbihdf -lblasr -lpbdata
make[1]: *** No rule to make target `DB.o', needed by `dazcon'.  Stop.
make[1]: Leaving directory `/home/kzhou/src/metag/pbdagcon/src/cpp'
make: *** [cpp] Error 2

A lots more progress than before:
configure.py --no-pbbam --gtest --sub
hou/src/metag/pbdagcon/DALIGNER/align.c
/home/kzhou/src/metag/pbdagcon/DALIGNER/align.c:61:16: fatal error: DB.h: No such file or directory
 #include "DB.h"
                ^
compilation terminated.
make[1]: *** [align.o] Error 1
make[1]: Leaving directory `/home/kzhou/src/metag/pbdagcon/src/cpp'
make: *** [cpp] Error 2

So the problem is not being able to find DB.h

make init-submodule 
run successfully,
Entering directory `/home/kzhou/src/metag/pbdagcon/src/cpp'
cc -O3 -Wall -Wextra -fno-strict-aliasing -MMD -MP -I  -I/home/kzhou/src/metag/pbdagcon/DAZZ_DB -I/home/kzhou/src/metag/pbdagcon/DALIGNER -I/home/kzhou/src/metag/pbdagcon/blasr_libcpp/alignment -I/home/kzhou/src/metag/pbdagcon/blasr_libcpp/pbdata -isystem.//third-party  -c -o align.o /home/kzhou/src/metag/pbdagcon/DALIGNER/align.c
/home/kzhou/src/metag/pbdagcon/DALIGNER/align.c:61:16: fatal error: DB.h: No such file or directory
 #include "DB.h"

I think that the make is missing the include directory to DB.h

--- a/configure.py
+++ b/configure.py
@@ -153,7 +153,7 @@ def compose_defines_pacbio(envin):
             'HDF5_INCLUDE', 'HDF5_LIB', 'HDF5_LIBFLAGS',
             'PBBAM_INCLUDE', 'PBBAM_LIB', 'PBBAM_LIBFLAGS',
             'HTSLIB_INCLUDE', 'HTSLIB_LIB', 'HTSLIB_LIBFLAGS',
-            'BOOST_INCLUDE',
+            'BOOST_INCLUDE','PTHREAD_LIBFLAGS',
             'ZLIB_LIB', 'ZLIB_LIBFLAGS',
             'DAZZ_DB_SRC', 'DAZZ_DB_INCLUDE',
             'DALIGNER_SRC', 'DALIGNER_INCLUDE',
@@ -205,6 +205,7 @@ def set_defs_defaults(env, nopbbam):
         'DALIGNER_SRC': os.path.join(ROOT, '..', 'DALIGNER'),
         'DAZZ_DB_INCLUDE': '${DAZZ_DB_SRC}',
         'DALIGNER_INCLUDE': '${DALIGNER_SRC}',
+        'PTHREAD_LIBFLAGS': '-lpthread',
     }
     pbbam_defaults = {
         'LIBPBIHDF_LIBFLAGS': '-lpbihdf',

